### DISCUSSION TITLE
Create the temp VM in the temp RG in a private network

### DISCUSSION THREAD

**Discussion by vamshisiram** [actions/virtual-environments#7362] [General questions]
I would like to use the generate image resource script to create the Image builder VM (temp VM in temp RG) in a private network in a virtual network.

Reading the instructions provided [here](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md#additional), it is not clear as to how to achieve this.

Do i have to modify the script .\helpers\GenerateResourcesAndImage.ps1 and add the variables virtual_network_name, virtual_network_resource_group_name, and virtual_network_subnet_name to the packer section?

![image](https://user-images.githubusercontent.com/90816346/228366082-e70fbc1f-8804-4118-a2be-beb54df70db8.png)

Or

Do i just pass in the variables virtual_network_name, virtual_network_resource_group_name, and virtual_network_subnet_name to the command when calling the function GenerateResourcesAndImage without adding the variables to the script block above?

tagging @Alexey-Ayupov @mikhailkoliada @igorboskovic3 @erik-bershel

**Comment by erik-bershel** [comment]
Hello @vamshisiram!
As you can see [here](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md#user-variables) there are a lot of variables which you can use by manual triggering packer or with using of predefined values via changing config file. The same as in your case. So you shouldn't just pass these variables to the command when calling the function from a script, it wouldn't work.
